crates/relayer/src/worker/map.rs

```diff
@@ -124,6 +124,10 @@ impl WorkerMap {
         dst: Chain,
         config: &Config,
     ) -> &WorkerHandle {
+
+        // Clear workers, to prevent returning stopping workers.
+        self.clean_stopped_workers();
+
         if self.workers.contains_key(&object) {
             &self.workers[&object]
```

Comment:
What do you think about checking and clearing only the specific worker instead of iterating through all of them?

```suggestion
        if self.workers.contains_key(&object) {
            if self.workers[&object].shutdown_stopped_tasks() {
                self.remove_stopped(self.workers[&object].0, self.workers[&object].1);
                let worker = self.spawn_worker(src, dst, &object, config);
                self.workers.entry(object).or_insert(worker)
            } else {
                &self.workers[&object]
            }
```

Reply:
Thanks a lot!, it seems good. I will apply this change :)